### ISSUE TITLE
Poor support for titles using multiple executables

### ISSUE THREAD

**Issue by LukeUsher** [LukeUsher/Cxbx-Reloaded#4]
Lack of support for XLaunchNewImage & HalReturnToFirmware prevents games using multiple executable files and games which reboot the current executable to switch game mode (eg. Buffy) from working.

At a minimum, an Xbe loader has to be implemented for XLaunchNewImage to work.

A more complete approach would be to properly implement HalReturnToFirmware and MmPersistContiguousMemory, allowing titles to reboot, launch other titles or even launch the Dashboard without patching XLaunchNewImage

**Comment by PatrickvL** [comment]
Also, when loading images, the base image address should be the same as on xbox. To assure this in Dxbx, I used a DLL for the actual emulator code, and used the Dxbx.exe executable as a placeholder.

```
// Reserve a block of Virtual Memory with a size equal the total memory
// available in an Xbox : 128 MB. This block has no predefined contents,
// and thus takes virtually no space in the EXE. This block makes sure that
// the whole 128MB range of Virtual Memory (from $00010000 up to $0800000)
// isn't occupied with anything else, and this fully available to our emulator.
const
  XBOX_MEMORY_SIZE = 128*1024*1024;
var
  Data: array [0..XBOX_MEMORY_SIZE-1] of Byte;

// Make sure this EXE gets loaded into the Xbox-specific Virtual Address Space :
{$IMAGEBASE $00010000} // = XBE_IMAGE_BASE

// Make sure we can't be loaded to another location either, by removing
// the relocation table. These two settings are the only way we know of,
// to reserver the Virtual Memory Range that the Xbox normally uses...
// This also removes the relocation table (and thus generates smaller executables)
// (See http://hallvards.blogspot.com/2006/09/hack12-create-smaller-exe-files.html)
{$SetPEFlags 1} // = Windows.IMAGE_FILE_RELOCS_STRIPPED

// Here, we create a static dependance on our actual emulation DLL,
// by importing the function that takes over control and start emulation :
procedure DxbxMain(const aData: Pointer; const aSize: Cardinal); stdcall; external 'DxbxKrnl';

```

**Comment by LukeUsher** [comment]
Solved in 1d89b04573025e265aa3978473f6ae6943c0c5f3 (via XLaunchNewImage)

MmPersistContiguousMemory and HalReturnToFirmware will be implemented later, it is overkill for now
